### DISCUSSION TITLE
Creating a branch in a fork

### DISCUSSION THREAD

**Discussion by davidstaheli** [community/community#19097] [Other Feature Feedback, Questions, & Ideas]
Make it easier to create a branch in a fork

**Comment by theteladras** [comment]
@davidstaheli 
How easier, what do you have in mind?

In my opinion it is easy enough so far, just by checking out the branches through the forked repo you do have an option of creating a new branch.

**Reply by xmo-odoo** [reply]
The experience is really sub-par when trying to edit in a fork via the Github UI, if the fork is not yours (not in your personal account), especially if it had to lock / disable / destroy matching branches to avoid user errors.

In that case there is really no way to create an up-to-date edition branch into the fork:

- in the fork you can only create branches from the fork's own branches, since the fork's matching branches were destroyed, they can't be synched (and they might be protected preventing you from synching them anyway)
- when editing on the source repository, you can not select an organisation fork, it will either write into the repository if you have write access, or in your personal fork if you don't

**Reply by miguellealperez** [reply]
yes.

**Reply by P123838** [reply]
嗨

**Comment by admuresan** [comment]
changing the source branch in drop down seems not to work, the source branch is always master

**Reply by wadevries** [reply]
To be more precise: the source branch is always the 'default branch', whatever source branch you select.

**Reply by wcorbeil1** [reply]
Same thing here

**Reply by pravincar** [reply]
This issue made the feature unusable for me

**Reply by jblanche** [reply]
Same bug here.

**Reply by AndyColbridgeAngstrom** [reply]
glad it's not just me.

**Reply by ansballard** [reply]
Hi all! This bug should be resolved, if you're still able to reproduce this issue definitely reach back out here.

**Reply by P123838** [reply]
> 大家好！這個錯誤應該得到解決，如果你仍然能夠重現這個問題，一定要回到這裡。

**Reply by masifmehmoodsami99** [reply]
> اس مسئلے نے اس خصوصیت کو میرے لیے ناقابل استعمال بنا دیا۔
Yes 🥰 thnkxx for the update

**Comment by Kazooo100** [comment]
Gives me an error saying I cant create branch from non fork.
very vague error
"Sorry the branch (name) could not be created"

**Comment by vshshjn7** [comment]
This should also support branch creation from the upstream tag.

**Reply by JamilGujjar** [reply]
Sir, I can't understand purpose of creating branch in fork. Please explain.

**Reply by brianjmurrell** [reply]
I think the _in fork_ part is a red-herring.  Ultimately, the problem is that the UX for creating a branch, anywhere, is horribly limited.  As is the UX for creating releases.

**Comment by oren-nonamesecurity** [comment]
The feature is broken, always creates the new branch as "master" and ignores the "Branch source" field

**Reply by ansballard** [reply]
Hi there! You should be able to use the branch source dropdown as intended after a fix yesterday, feel free to reach back out if it's still an issue.

**Reply by ronchipman19** [reply]
Testing this now !  So far so good.  I can tell right away that it isn't pulling from master/main based on the update tag associated with it.  Thanks for fixing this!

**Reply by P123838** [reply]
> 該功能已損壞，始終將新分支創建為“master”並忽略“Branch source”字段

**Comment by elongl** [comment]
Why can't I create a new branch from a tag?

**Reply by masifmehmoodsami99** [reply]
@

**Comment by shashank21j** [comment]
Allow creating a branch with an orphan option.

**Comment by jfullstackdev** [comment]
I'm not very sure of your errors but on my end there is none, I can switch easily from the two branches, I can create one, if I reload the page, it's not switching back to the default one

![image](https://user-images.githubusercontent.com/47092464/181001804-12c54104-1fe5-4488-a5f4-9f3065034603.png)

![image](https://user-images.githubusercontent.com/47092464/181001871-efe48c02-49e0-4343-9b4a-204266d1db7d.png)

maybe you installed extensions that alters that behaviour

**Comment by brianjmurrell** [comment]
Similar to the issues above with creating a branch from a tag, you cannot create a branch at any chosen git commit.

But honestly, using this branch creation functionality for me is because I am trying to work around the even more limiting factor that releases cannot be made from commit hashes or tags, but only branches and some (useless) value of "recent commit"s.  I cannot imagine who decided that releases are made from branches (only).

**Comment by ddzz** [comment]
Is it possible to disable this feature? It's causing issues with `${{ github.ref }}` in our GitHub Actions CI pipeline.

**Reply by ansballard** [reply]
Hi! As far as I know this shouldn't cause an issue with your CI pipeline, this is essentially a UI change on the branches page for forked repos. I would create a new discussion with your issue and link to this one, so the appropriate teams can respond.

**Comment by spotlesscoder** [comment]
make it possible to create a new branch from a certain commit ID or tag as base

**Comment by komealy** [comment]
Is there a way to have the input box already active so I don't have to click twice? Just one click on the button, then start typing the new branch name would be amazing, then press `Enter` to make it happen.

**Comment by micsthepick** [comment]
make a branch from another sibling fork

**Reply by kaovilai** [reply]
Wholeheartedly agree.

**Comment by ajarzyn** [comment]
It would be grate to have possibility to create branch from base repository.
Base - > Fork1 -> MyFork

It is impossible to fork Base when there's fork from child repository (Fork1)
Currently only MyFork and Fork1 are allowed.

Impossible to create branch from Base repository.

Hope you feel it may be useful.
Best of luck.

**Comment by GetToSet** [comment]
When working on repositories which I'm one of the collaborator, I never add upstream repo as remote of my locally forked repo to prevent accident push to the upstream repo. So I use this feature at GitHub to copy a new upstream branch into the fork. I wish it could automatically filling the branch-name with the selected upstream branch or (at least) allow manually select-copy so that I don't need to copy-type the branch name and double-check if I typed correct.

**Comment by dougransom** [comment]
I would like to be able to branch at a specific commit.

**Comment by rvsfara** [comment]
Or allow creating from a part of the project as a model, or orm, because I only want to take advantage of this one

**Comment by carrved** [comment]
I would like to make a clean branch. (nothing in the new branch)

**Comment by Feliipe93** [comment]
Enable or make possible to  add branches from one repository to another

**Comment by ryanAtRite** [comment]
I'm noticing when I create a branch from the Branches page, it creates successfully but does not refresh the branches list.  If I refresh the page manually it shows up.

**Reply by allthedoll** [reply]
I tried to recreate this, but mine refreshes. Can you take a screen shot or let me know the browser you're using?  🙇🏼‍♀️

**Comment by DangerousTod** [comment]
Doing the same thing a new way for the third time.
It works, leave it alone.

**Comment by Appz-dk** [comment]
I would like to be able to simply create an empty branch to use for my gh-pages. But I am forced to choose a "Branch source".
I would just like a clean branch where I then can push my dist folder content to.

**Comment by rvil79** [comment]
> اجعل من السهل إنشاء فرع في مفترق
عند التحرير في مستودع المصدر ، لا يمكنك تحديد مفترق مؤسسة ، إما أن يكتب في المستودع إذا كان لديك حق الوصول للكتابة ، أو في مفترقك الشخصي إذا لم يكن لديك
.

**Comment by Chadders13** [comment]
Yer creating a branch from a tag would be a great addition.

**Comment by akshay-nm** [comment]
I'm getting a 500 when trying to create a new branch from master

**Comment by lndmanh** [comment]
I would just like a clean branch where I then can push my dist folder content to.

**Comment by tjamet** [comment]
It would be interesting to be able to create a branch providing a plain sha (for recovering an old commit for example)

<img width="437" alt="image" src="https://user-images.githubusercontent.com/6061440/229568865-93401999-574a-4761-b91d-c537d171da4e.png">

**Comment by vmari** [comment]
This new feature is great! I really appreciate the effort that went into making it user-friendly and intuitive.

One suggestion I have is to enable **autofocus** once the modal is open. This would greatly improve the user experience and make it even easier to navigate through the different options. 

It can be really frustrating when you're in a rush and the current behavior on a platform triggers background key shortcuts.

Overall, though, I think this is a fantastic addition to the platform and I look forward to using it more in the future.

**Comment by raymond-chetty** [comment]
Suggestions:
1. It might be worth not having a pull request box on a newly created master or other branch in the "Active" view. A "dangling" active branch? It doesn't look like the default branch is considered active, why are the others? (See image 1)
2. There might also be an issue if you fork a repository or template repository. (See image 2)
3. Add an information bubble somewhere on the page "View all branches" -> "Active" describing what [active means][1] (an "i" or "?" in a circle)


I'm not sure I want to recreate the issue, but after the creation of a codespace template, I generated devcontainer configuration within a codespace and then created a repository from it. I had a few issues when I was modifying the default branching system so that the main branch wasn't master or the codespace branch. There might also be an issue with syncing with git in the original codespace, trying to pull/fetch new branches from origin or something.

The main issue I saw was a dangling of the master branch in the active branch window. It looks like it is there for creating a pull request, but there were some issues, like there isn't any code to pull or the branch is up-to-date. 

I noticed this after doing the following... or close to it.

1. create two forks from the codespace named branch "funky-but-luvin" etc. master and dev (or just dev?).
5. set dev as the default branch
6. delete the codespace named branch
7. see "master" branch still as an active branch. see "dev" isn't there as an active branch.

Before doing this I had checked in a few changes, and the master and dev were on the same commit, I think.

[1]: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/viewing-branches-in-your-repository

1.
![screenshot1](https://user-images.githubusercontent.com/44623277/235785773-b3b0eccd-8f9c-4b0e-827a-509a22c1ed17.png)
2.
![screenshot2](https://user-images.githubusercontent.com/44623277/235787514-9d5dec57-fbc3-4f4e-ade1-bec9975d4006.png)

In this case(2) there isn't anything to compare, both branches are the same, but the repo says master has recent pushes in a notification like box and the pull request button shows all the files in the repository as new. I think it's showing the first commit.

**Reply by raymond-chetty** [reply]
I think "active" means non default branch. Maybe add it to the definition in the docs.

**Reply by raymond-chetty** [reply]
Some documentation about git workflows. https://www.atlassian.com/git/tutorials/comparing-workflows

+ GitHub flow https://docs.github.com/en/get-started/quickstart/github-flow

**Comment by mutleybg** [comment]
In "Create a branch" popup the "Branch source" dropdown is small. If the branch name is long you can not see it and there is no tooltip showing the whole branch name.
